Automattic/sensei
# Improve Learning Mode layout blocks in the site editor

This is a first step in making Learning Mode templates properly editable and flexible. There is still work to be done in follow-up PRs to properly decouple CSS and move as much as possible to the template, utilizing core block features like spacing and layout options. 

It also adds a basic version to load separate CSS based on the template version being used, mainly to keep the feature branch in a state where it does not immediately break the current template.

### Changes proposed in this Pull Request

- Introduce new blocks: Fixed Header, Sidebar, Two-Column Layout, Main Content, Spacer (Auto) (as variations of a hidden generic 'Interface Element' block). 
   — These are the building blocks of the current LM template. Previously, these were all Group blocks with custom classnames, but changing or removing the classes, or some container blocks, would break the layout. 
- Rebuild the learning mode template upon those blocks, and on new core blocks and layout options (like Row)
- Build a different CSS bundle for the new template. This should be followed up in #5541
- Add workaround to show sidebar and main content columns in 100% height in the editor 

### Testing instructions

Test old template:

- Check out `trunk`
- Customize the learning mode template in the Site Editor. 
- Check out this branch. Check that the customized template still looks as expected in the frontend
- Clear customizations for the template in the site editor. This should switch to the new one.
- Check that this also looks good both in editor and frontend


### Screenshot / Video

| Before | After |
| - | - |
| ![image](https://user-images.githubusercontent.com/176949/184664334-1b7ea8a5-30c7-492f-af7e-92a4e8475af7.png) | ![image](https://user-images.githubusercontent.com/176949/184664264-1ce9d287-15b1-47d0-8a62-1864a67a773b.png) |

| Before | After |
| - | - |
| <img width="244" alt="image" src="https://user-images.githubusercontent.com/176949/185078428-ce1f035a-1196-40a9-b051-856724471582.png"> | <img width="270" alt="image" src="https://user-images.githubusercontent.com/176949/185109944-4d2935ec-9e19-42a2-aaf9-650082d86324.png"> |
